Allow for merging of SCEs by integration group

cbethell opened the pull request:
**Issue Addressed**
Closes #340

**What is the purpose of these changes?**
<!--Provide some background on the changes proposed.-->
The purpose of this PR is to allow users to specify which integration group each SCE belongs to (e.g. libraries from the same disease are merged together rather than all libraries being merged together into one).

**What changes did you make?**
<!--Describe the concrete changes that you made or additional features that were added, be as detailed as possible in the steps you took.-->
The following modifications were made to the `merge-sce.R` script in the data integration module:
- an argument to take the desired `integration_group`
- a step to filter in the input metadata on that integration group argument

These modifications allow users to indicate which libraries (SCEs) should be grouped together.

The example metadata file was also updated to included an `integration_group` column and filled in with generic values.

**Any comments, concerns, or questions important for reviewers**
Does this PR see to adequately address #340?

**Checklist**

Place an `x` in all boxes that you have completed.

- [x] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [ ] I have added all necessary documentation (if applicable)

allyhawkins commented on optional-integration-analysis/merge-sce.R:
```diff
@@ -72,6 +77,15 @@ if(is.null(opt$input_metadata_tsv)){
 
+# Check that integration group is provided
+if(is.null(opt$integration_group)){
+  stop("The name of the corresponding integration group is missing.")
```
```suggestion
  stop("The `integration_group` is missing and required for identifying which objects should be merged together.")
```
Just clarifying the error message a little bit.

allyhawkins commented on optional-integration-analysis/merge-sce.R:
```diff
@@ -72,6 +77,15 @@ if(is.null(opt$input_metadata_tsv)){
+} else {
+  # Filter the metadata to include only information relevant to the specified integration group
+  input_metadata <- input_metadata %>%
+    dplyr::filter(integration_group == opt$integration_group)
```
Before we get here we should add a check that `opt$integration_group` is in `input_metadata$integration_group` first.

allyhawkins reviewed (commented):
Just a few minor comments that should be addressed.

cbethell commented:
Thanks for the review @allyhawkins! I added in that extra check and removed the additional ids from the example metadata as suggested -- let me know what you think now.
